app/models/core_data_connector/media_content.rb

```diff
@@ -10,6 +10,31 @@ class MediaContent < ApplicationRecord
     include TripleEyeEffable::Resourceable
     include UserDefinedFields::Fieldable
 
+    after_save :update_manifests
+
+    def update_manifests
+      self.related_relationships.each do |relationship|
+        if relationship.primary_record_type == 'CoreDataConnector::MediaContent'
+          model_class = relationship.related_record_type.constantize
+        else
+          model_class = relationship.primary_record_type.constantize
+        end
+
+        if relationship.primary_record_id == self.id
```

Comment:
I think this logic could be problematic. Since `primary_record` is a polymorphic relationship, it's possible that the two records could have the same `id` value, but be of different classes.

Could we set the `related_record_id` within the same if/else block above where we're checking the `primary_record_type`?